soh/soh/Enhancements/mods.cpp

```diff
@@ -1223,6 +1224,22 @@ void RegisterToTMedallions() {
     });
 }
 
+void RegisterFloorSwitchesHook() {
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnActorInit>([](void* refActor) {
+        Actor* actor = static_cast<Actor*>(refActor);
+        if (actor->id != ACTOR_OBJ_SWITCH || !CVarGetInteger("gEnhancements.FixFloorSwitches", 0)) {
+            return;
+        }
+
+        ObjSwitch* switchActor = reinterpret_cast<ObjSwitch*>(actor);
+        s32 type = (switchActor->dyna.actor.params & 7);
+
+        if (switchActor->dyna.actor.params == 4608 || switchActor->dyna.actor.params == 14848) {
```

Comment:
could this be in hex?
```suggestion
        if (switchActor->dyna.actor.params == 0x1200 || switchActor->dyna.actor.params == 0x3A00) {
```
not sure we have any macros for these, but since bitwise operations are done on params, it's generally better to keep them as hex